pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -131,6 +131,43 @@ const (
 	retrySleepTime = 20 * time.Millisecond
 )
 
+const (
+	// These labels MUST match what defined in pkg/kubelet/apis/well_known_labels.go
+	labelOSBeta   = "beta.kubernetes.io/os"
+	labelArchBeta = "beta.kubernetes.io/arch"
+)
+
+// labelReconcileInfo lists Node labels to reconcile, and how to reconcile them.
+// primaryKey and secondaryKey are keys of labels to reconcile.
+//   - If both keys exist, but their values don't match. Use the value from the
+//   primaryKey as the source of truth to reconcile.
+//   - If ensureSecondaryExists is true, and the secondaryKey does not
+//   exist, secondardayKey will be added with the value of the primaryKey.
+var labelReconcileInfo = []struct {
+	primaryKey            string
+	secondaryKey          string
+	ensureSecondaryExists bool
```

Comment:
if `ensureSecondaryExists` is only one value in this PR, prefer to remove it for now; and add it back when there're other cases.

Reply:
We are going to use this mechanism, and I think leaving this actually makes the code more readable when we switch in the future.

If you're okay, I'd rather leave this as it is.

Reply:
that's ok